### PR TITLE
Feat: Auth context provider

### PR DESCRIPTION
**Pull request by joao-conde**
| - | - |
| --- | --- |
| Issue | Originated by https://github.com/ripe-tech/ripe-id-mobile/issues/3 |
| Dependencies | https://github.com/ripe-tech/ripe-id-api-js/pull/5 |
| Decisions | Create a simpler login context without the whole OAuth workflow. |

### CODE REVIEW

File: react/components/contexts/auth-context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    async login(email, password) {
+        const response = await this.ripeIdApi.login(email, password);
+        verify(response.sid, "No session ID found", 404);
+        verify(response.username, "No username found", 404);
+        verify(response.tokens, "No tokens list found", 404);
+
+        this.ripeIdApi.sessionId = response.sid;
+        this.setState({
+            account: response.username,
```
Comment (BeeMargarida) [inline react/components/contexts/auth-context/auth-context.js:44]: To make it similar to the oAuthContext, it would be nice to populate the account with the account object and not the username. Is it possible? Or at least call this username instead of account.

Comment (joao-conde) [reply react/components/contexts/auth-context/auth-context.js:44]: The response to login has: `sid` a string, `username` a string too, and `tokens` a list of strings. Should I save it all under account? I think on the other context, account is not the same. I could call `ripeIdApi.selfAccount` and store it tho. Just an extra call but it would mean both contexts have the same data.

Comment (BeeMargarida) [reply react/components/contexts/auth-context/auth-context.js:44]: Yap, that would be nice. Or remove the account and just add username

File: react/components/contexts/auth-context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        await AsyncStorage.setItem("account", JSON.stringify(account));
+        await AsyncStorage.setItem("tokens", JSON.stringify(response.tokens));
+        await this.trySetAccount();
+    }
+
+    async logout(message = null) {
+        // re-creates the RIPE ID API instance and invalidates the
+        // current RIPE SDK instance as it's no longer going to be used
+        this.ripeIdApi = new RipeIdAPI(this.props.options);
+        if (this.context.ripeApi) this.context.ripeApi.unauth();
```
Comment (BeeMargarida) [inline react/components/contexts/auth-context/auth-context.js:72]: Is this valid in this context now?

Comment (joao-conde) [reply react/components/contexts/auth-context/auth-context.js:72]: Nope :+1:

Comment (BeeMargarida) [reply react/components/contexts/auth-context/auth-context.js]: Does it also need to recreate the ripe ID?

Comment (joao-conde) [reply react/components/contexts/auth-context/auth-context.js]: Think so, if you logout you basically want to erase every token, the approach on  the oauth context is to recreate the api object (clears session and auth tokens)

Comment (BeeMargarida) [reply react/components/contexts/auth-context/auth-context.js]: Yeah, you are right

**Review by BeeMargarida — APPROVED** [review]

**Review by 3rdvision — APPROVED** [review]

File: react/components/contexts/auth-context/auth-context.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.ripeIdApi = new RipeIdAPI(this.props.options);
+        this.state = {
+            loaded: false,
+            logoutMessage: null,
```
Comment (gcandal) [inline react/components/contexts/auth-context/auth-context.js:30]: Why is this needed?

**Review by gcandal — COMMENTED** [review]
Delaying the merge until contexts are ported to https://github.com/ripe-tech/ripe-commons-native

File: react/components/contexts/auth-context/auth-context.js (hunk 3 above)
Comment (joao-conde) [reply react/components/contexts/auth-context/auth-context.js:30]: I replicated from the `OAuthContext`. It is used here: https://github.com/ripe-tech/ripe-id-mobile/blob/master/app.js#L139
